docs/specifications/tests/Syntax-TP/syntax06.md

```diff
@@ -1,54 +1,134 @@
-## SYNTAX06: No illegal characters in the SOA RNAME field
+# SYNTAX06: No illegal characters in the SOA RNAME field
 
-### Test case identifier
-**SYNTAX06** There must be no illegal characters in the SOA RNAME field
+## Test case identifier
+**SYNTAX06**
 
-### Objective
+## Objective
 
 The SOA RNAME field is a mailbox address. The SOA RNAME field is defined
-in section 3.3 of [RFC 1034](https://tools.ietf.org/rfc/rfc1034.txt) and
-section 2.2 of [RFC 1912](https://tools.ietf.org/rfc/rfc1912.txt). The RNAME
-field should follow the rules of an e-mail address also defined in section
-3.4.1 of [RFC 2822](https://tools.ietf.org/html/rfc2822#section-3.4).
+in section 3.3.13 of [RFC 1035] and section 2.2 of [RFC 1912]. The RNAME
+field should follow the rules of an e-mail address also defined in 
+[RFC 5322, section 3.4.1].
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Ordered description of steps to be taken to execute the test case
+
+1. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+2. Create a SOA query for the apex of the *Child Zone* with RD flag unset.
+
+3. For each name server in *Name Server IP* do:
+   1. Send the SOA query over UDP to the name server.
+   2. If the name server does not respond with a DNS response, then:
+      1. Output *[NO_RESPONSE]*.
+      2. Go to next server.
+   3. If the DNS response does not include a SOA record in the 
+      answer section, then:
+      1. Output *[NO_RESPONSE_SOA_QUERY]*.
+      2. Go to next server.
+   4. Extract the RNAME from the SOA record.
+   5. Convert the first "." without backslash quoting to an "@" in 
+      the RNAME.
+   6. Convert any backslash quoted "." to a single "." without quoting
+      (see section 5.1, 5.3 and 8 in [RFC1035] for the use of backslash).
+   7. If the converted string ("Mail address") does not meet the 
+      mail address specification in [RFC 5322, section 3.4.1], then 
+      1. Output *[RNAME_RFC822_INVALID]*.
+      2. Go to next server.
+   8. Extract the domain part (to the right of "@") from the *Mail 
+      address* ("Domain Part").
+   9. Create an MX query for the *Domain Part* and do a [DNS Lookup]
+      of that query. 
+   10. If the lookup of MX does not return a DNS response with RCODE 
+       "NOERROR", then:
+       1. Output *[RNAME_MAIL_DOMAIN_INVALID]*.
+       2. Go to next server.
+   11. If the MX lookup returned a CNAME or a chain of CNAMEs then
+       set *Domain Part* to be the the final name instead of the
+       domain from the email address.
```

Comment:
email address -> \*Mail address\*?

Maybe even put the "instead of"part within parentheses?

Reply:
Adjusted.